app/models/message.rb

```diff
@@ -65,17 +67,41 @@ def as_json(options = {})
       :impacted_services => self.impacted_services,
       :created_at => self.created_at.strftime("%A, %B %d, %Y at %l:%M %p"),
       :created_at_in_words => time_ago_in_words(self.created_at) + ' ago',
-      :recipient_count =>
-        if self.log
-          self.log.recipient_count ? self.log.recipient_count.to_s + " (" + self.log.viewed_count.to_s + " viewed)" : 'Calculating'
+      :publishers => self.publishers,
+      :recipient_counts =>
+        if self.logs
+          self.logs.map do |log|
+            {
+              :publisher => log.publisher ? log.publisher.name : 'E-mail',
+              :count => log.recipient_count || 'Unavailable',
+              :viewed => log.viewed_count
+            }
+          end
         else
-          'Unavailable'
+          []
         end,
-      :send_status =>
-        if self.log
-          self.log.send_status.to_s.capitalize
+      :send_statuses =>
+        if self.logs
+          self.logs.map do |log|
+            {
+              :publisher => (log.publisher ? log.publisher.name : 'E-mail'),
+              :status =>
+#                if log.send_status.to_s.capitalize == 'Queued' && 
```

Comment:
These commented-out lines are a possible fix for notifying users whether or not delayed jobs are running. They're commented out because I don't know if there's a delayed_job.pid file on the server/whether or not delayed jobs are started from scripts/delayed_job.